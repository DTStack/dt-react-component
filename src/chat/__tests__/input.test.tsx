import React from 'react';
import { act, cleanup, fireEvent, render } from '@testing-library/react';
import { input } from 'ant-design-testing';

import Input from '../input';

describe('Test Chat Input', () => {
    beforeAll(() => {
        window.ResizeObserver = jest.fn().mockImplementation(() => ({
            disconnect: jest.fn(),
            observe: jest.fn(),
            unobserve: jest.fn(),
        }));
    });

    beforeEach(cleanup);
    it('Match snapshot', () => {
        expect(render(<Input />).asFragment()).toMatchSnapshot();
    });

    it('Should call onChange', () => {
        const onChange = jest.fn();
        const { container } = render(<Input onChange={onChange} />);
        act(() => {
            input.textarea.fireChange(container, 'test');
        });
        expect(onChange).toBeCalledWith('test');
    });

    it('Should call specify function', () => {
        const onPressShiftEnter = jest.fn();
        const onChange = jest.fn();
        const onPressEnter = jest.fn();
        const { container } = render(
            <Input
                value="test"
                onPressShiftEnter={onPressShiftEnter}
                onChange={onChange}
                onPressEnter={onPressEnter}
            />
        );
        // Simulate press shift + enter
        // FIXME: ant-design-testing 不支持触发 shift 的 onPressEnter 事件
        const ele = container.querySelector<HTMLTextAreaElement>('textarea')!;
        fireEvent.keyDown(ele, { keyCode: 13, shiftKey: true });

        expect(onPressShiftEnter).toBeCalled();
        expect(onChange).lastCalledWith('test\n');
        expect(onPressEnter).not.toBeCalled();

        onChange.mockClear();
        onPressShiftEnter.mockClear();
        // Simulate press enter
        input.textarea.firePressEnter(container);
        expect(onPressShiftEnter).not.toBeCalled();
        expect(onChange).not.toBeCalled();
        expect(onPressEnter).toBeCalledTimes(1);
    });

    it('Should call onSubmit', () => {
        const onSubmit = jest.fn();
        const { container, rerender } = render(<Input onSubmit={onSubmit} />);
        const ele = container.querySelector<HTMLSpanElement>('.dtc__chat__textarea__send')!;
        fireEvent.click(ele);
        expect(onSubmit).toBeCalled();

        onSubmit.mockClear();
        rerender(<Input button={{ disabled: true }} onSubmit={onSubmit} />);
        fireEvent.click(ele);
        expect(onSubmit).not.toBeCalled();
    });
});
