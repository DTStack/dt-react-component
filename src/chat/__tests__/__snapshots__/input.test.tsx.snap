// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Chat Input Match snapshot 1`] = `
<DocumentFragment>
  <div
    class="dtc__chat__textarea__container"
  >
    <textarea
      class="ant-input dtc__chat__textarea"
    />
    <span
      class="dtc__icon dtc__chat__textarea__send"
    >
      <svg
        fill="none"
        height="1em"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="M17.999 3.858 3.35 9.204a1.66 1.66 0 0 0-1.095 1.458 1.66 1.66 0 0 0 .862 1.602l4.454 2.44c.27.143.593.098.808-.118l.737-.72 4.508-4.428a.677.677 0 0 1 .952.018c.26.261.26.693-.009.954l-4.562 4.473-.61.603a.68.68 0 0 0-.118.828l2.64 4.626c.315.513.845.81 1.429.81.063 0 .117 0 .18-.009a1.68 1.68 0 0 0 1.41-1.107L20.153 6a1.69 1.69 0 0 0-.404-1.764 1.64 1.64 0 0 0-1.751-.378"
          fill="url(#paint0_linear_3878_6538)"
          fill-rule="evenodd"
        />
        <defs>
          <lineargradient
            gradientUnits="userSpaceOnUse"
            id="paint0_linear_3878_6538"
            x1="2.25"
            x2="23.921"
            y1="3.75"
            y2="11.552"
          >
            <stop
              stop-color="#00BAC6"
            />
            <stop
              offset="0.5"
              stop-color="#0067FF"
            />
            <stop
              offset="1"
              stop-color="#450FDE"
            />
          </lineargradient>
        </defs>
        <defs>
          <lineargradient
            gradientUnits="userSpaceOnUse"
            id="paint0_linear_3878_6538_hover"
            x1="2.25"
            x2="23.921"
            y1="3.75"
            y2="11.552"
          >
            <stop
              stop-color="#08C4FF"
            />
            <stop
              offset="0.5"
              stop-color="#4892FF"
            />
            <stop
              offset="1"
              stop-color="#8A61FF"
            />
          </lineargradient>
        </defs>
      </svg>
    </span>
  </div>
</DocumentFragment>
`;
